### PR TITLE
fix: improve hover text 'unknown' for better UX.

### LINKED ISSUE #38: [create] "balance computation failed" improve hover text
After creating a request, "Unknown" with hover text "The balance computation might have failed" is shown. "Unknown" means that the request has not yet been persisted to IPFS. Thus, I think the text could be improved to help the user understand what is happening.

image

### PR DESCRIPTION
**Pull request by PijiDev**
Fixes #38

### CODE REVIEW

File: packages/shared/src/index.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export type RequestStatus =
   | "pending"
   | "canceled"
   | "overpaid"
-  | "unknown";
```
Comment (MantisClone) [inline packages/shared/src/index.tsx:10]: @PijiDev Please revert this. This change doesn't successfully build.

```console
mantisclone@mantisclone-Inspiron-7620:~/projects/request-apps-piji$ yarn workspaces run build
yarn workspaces v1.22.19

> request-shared
yarn run v1.22.19
$ tsc
src/helpers/parseRequest.ts:75:5 - error TS2322: Type '"unknown" | "canceled" | "paid" | "overpaid" | "pending" | "open"' is not assignable to type 'RequestStatus'.
  Type '"unknown"' is not assignable to type 'RequestStatus'.

75     status,
       ~~~~~~

  src/index.tsx:18:3
    18   status: RequestStatus;
         ~~~~~~
    The expected type comes from property 'status' which is declared here on type 'IParsedRequest'


Found 1 error.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 2
Command: /home/mantisclone/.nvm/versions/node/v18.12.1/bin/node
Arguments: /home/mantisclone/.nvm/versions/node/v18.12.1/lib/node_modules/yarn/lib/cli.js run build
Directory: /home/mantisclone/projects/request-apps-piji/packages/shared
Output:

info Visit https://yarnpkg.com/en/docs/cli/workspaces for documentation about this command.
```

File: packages/request-ui/src/RStatusBadge.tsx [hunk 2]
```diff
@@ -40,8 +40,8 @@ export const RStatusBadge = (props: IProps) => {
       <div
         className={[classes.status, props.className].join(' ')}
         title={
-          props.status === 'unknown'
```
Comment (MantisClone) [inline packages/request-ui/src/RStatusBadge.tsx:43]: Please also revert this line from 'waiting' to 'unknown'

**Review by MantisClone — CHANGES_REQUESTED** [review]

File: packages/request-ui/src/RStatusBadge.tsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ export const statusLabels: Record<RequestStatus, string> = {
   pending: 'Pending',
   canceled: 'Canceled',
   overpaid: 'Overpaid',
-  unknown: 'Unknown',
+  waiting: 'Waiting',
```
Comment (MantisClone) [inline packages/request-ui/src/RStatusBadge.tsx:27]:
```suggestion
  waiting: 'Creating Request ',
```

File: packages/request-ui/src/RStatusBadge.tsx [hunk 4]
```diff
@@ -40,8 +40,8 @@ export const RStatusBadge = (props: IProps) => {
       <div
         className={[classes.status, props.className].join(' ')}
         title={
-          props.status === 'unknown'
-            ? 'The balance computation might have failed'
+          props.status === 'waiting'
```
Comment (MantisClone) [inline packages/request-ui/src/RStatusBadge.tsx:43]:
```suggestion
          props.status === 'unknown'
```

File: packages/shared/src/index.tsx [hunk 5]
```diff
@@ -7,7 +7,7 @@ export type RequestStatus =
...
+  | "waiting";
```
Comment (MantisClone) [inline packages/shared/src/index.tsx:10]: @PijiDev Please revert this. This change doesn't successfully build.

```suggestion
  | "unknown";
```

```console
mantisclone@mantisclone-Inspiron-7620:~/projects/request-apps-piji$ yarn workspaces run build
yarn workspaces v1.22.19

> request-shared
yarn run v1.22.19
$ tsc
src/helpers/parseRequest.ts:75:5 - error TS2322: Type '"unknown" | "canceled" | "paid" | "overpaid" | "pending" | "open"' is not assignable to type 'RequestStatus'.
  Type '"unknown"' is not assignable to type 'RequestStatus'.

75     status,
       ~~~~~~

  src/index.tsx:18:3
    18   status: RequestStatus;
         ~~~~~~
    The expected type comes from property 'status' which is declared here on type 'IParsedRequest'


Found 1 error.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed.
Exit code: 2
Command: /home/mantisclone/.nvm/versions/node/v18.12.1/bin/node
Arguments: /home/mantisclone/.nvm/versions/node/v18.12.1/lib/node_modules/yarn/lib/cli.js run build
Directory: /home/mantisclone/projects/request-apps-piji/packages/shared
Output:

info Visit https://yarnpkg.com/en/docs/cli/workspaces for documentation about this command.
```

**Review by MantisClone — CHANGES_REQUESTED** [review]

**Comment by PijiDev** [comment]
The build now works, for now I kept the ` ? 'The balance computation might have failed'`, because the original code expects a conditional string.

File: packages/request-ui/src/RStatusBadge.tsx [hunk 6]
```diff
@@ -40,8 +40,8 @@ export const RStatusBadge = (props: IProps) => {
...
+          ? 'The balance computation might have failed'
```
Comment (MantisClone) [inline packages/request-ui/src/RStatusBadge.tsx:44]:
```suggestion
          ? 'Waiting for request to be stored in IPFS'
```

**Review by MantisClone — CHANGES_REQUESTED** [review]
Thanks for taking a second look at this.  I like that you changed the state name from `unknown` to `waiting`. However, I think the hover text can still be improved.

**Review by MantisClone — APPROVED** [review]

**Pull request merged**